### ISSUE TITLE
FileImage unable to load image from disk, but MemoryImage can load it

### ISSUE THREAD

**Issue by buaashuai** [flutter/flutter#26277]
Hi, i meet a confusing problem.

I have an image on disk, its path  is `'/data/user/0/com.buaa.ateducation/app_flutter/9d682710eff4f8d68f1b933b7fa249b3/custom_task/defaultStudent/TXUGC_20190109_153124872.jpg'`.  The file is really exist.  And when i use Method A to show it in flutter, it cannot be displayed. But when i use Method B to show it in flutter, it displays correctly.

When i use Method A, my Android Studio has no errors or warnings in the console panel. Maybe there is some  problem with **FileImage** in flutter.

**Method A:**
```dart
String imagePath = '/data/user/0/com.buaa.ateducation/app_flutter/9d682710eff4f8d68f1b933b7fa249b3/custom_task/defaultStudent/TXUGC_20190109_153124872.jpg';
ImageProvider imageProvider = FileImage(File(imagePath));
……
Container(
            decoration: BoxDecoration(
                image: DecorationImage(
              image: imageProvider,
              fit: BoxFit.cover,
            )),
),
```

**Method B:**
```dart
String imagePath = '/data/user/0/com.buaa.ateducation/app_flutter/9d682710eff4f8d68f1b933b7fa249b3/custom_task/defaultStudent/TXUGC_20190109_153124872.jpg';

ImageProvider imageProvider = new MemoryImage(loadData());

Uint8List loadData() {
    File file = File(imagePath);
    Uint8List bytes = file.readAsBytesSync();
    return bytes;
  }

……
Container(
            decoration: BoxDecoration(
                image: DecorationImage(
              image: imageProvider,
              fit: BoxFit.cover,
            )),
),
```

**Flutter Doctor**
```
[✓] Flutter (Channel dev, v1.0.0, on Mac OS X 10.14.1 18B75, locale zh-Hans-CN)
[!] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    ✗ Android license status unknown.
[!] iOS toolchain - develop for iOS devices (Xcode 10.0)
    ✗ Verify that all connected devices have been paired with this computer in Xcode.
      If all devices have been paired, libimobiledevice and ideviceinstaller may require updating.
      To update with Brew, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew uninstall --ignore-dependencies usbmuxd
        brew install --HEAD usbmuxd
        brew unlink usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    ✗ ios-deploy out of date (1.9.4 is required). To upgrade with Brew:
        brew upgrade ios-deploy
[✓] Android Studio (version 3.2)
[!] IntelliJ IDEA Ultimate Edition (version 2018.2)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[!] VS Code (version 1.30.1)
[✓] Connected device (1 available)

! Doctor found issues in 4 categories.
```

**Comment by zoechi** [comment]
>  it cannot be displayed

It just does not show? No error/exception output?

**Comment by zoechi** [comment]
Can you create a minimum reproduction and publish it as GitHub repo?

**Comment by buaashuai** [comment]
> Can you create a minimum reproduction and publish it as GitHub repo?

I made a flutter plugin. It has a native video recorder activity. It is opened by a channel method using `activity.startActivityForResultAfter`. After the activity record a new video, it will close and pass a video cover path to the plugin and then the plugin passes the video cover path to the flutter. Then, the flutter will display the video thumbnail. Sometimes, the image cannot be displayed when i using **Method A** , I think maybe **FileImage** needs to execute some synchronizing steps before the file is ready for reading.

**Comment by quangduy90** [comment]
I have the same problem when I try to use FileImage, but maybe just for IOS, it works well on android.
Anyone can fix that ?

**Comment by kkacquah** [comment]
I have the same problem for android

**Comment by salime45** [comment]
I have the same problem

**Comment by atul-chaudhary** [comment]
the same problem is occurring
```

════════ Exception caught by image resource service ════════════════════════════════════════════════
The following FileSystemException was thrown resolving an image codec:
Cannot open file, path = '/storage/emulated/0/Download/denphy walls/cmprDigit (160).jpg.jpg/' (OS Error: Not a directory, errno = 20)

```

My code for opening file is:-
```
ClipRRect(
                  borderRadius: BorderRadius.circular(8.0),
                  child: Image.file(File(imgPath))
                ),

```

**Comment by opringle** [comment]
I am also experiencing this issue loading file images from temp dir on android

**Comment by ariefannur** [comment]
got same issue can't load local tmp image file `Cannot open file, path ={path} (OS Error: No such file or directory, errno = 2)`

**Comment by NUS-Anonymous** [comment]
anyone has solution for this yet?

**Comment by pedromassangocode** [comment]
Hi @buaashuai  and anyone else facing this issue
Can you please provide your `flutter doctor -v`, your `flutter run --verbose` and a minimal complete reproducible code sample?
Thank you

**Comment by goderbauer** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. We are therefore reluctantly going to close this bug for now. Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!Thanks for your contribution.

**Comment by Ahmad-Hayek** [comment]
I've been working on this for more than 3 days.
What is this rubbish called Flutter??
Shall we stay the whole life looking for something that can be solved in 5 seconds in Java?

**Comment by Warpath** [comment]
I also encountered this problem. The pictures downloaded to the private directory of the Android phone cannot be loaded with FileImage, but the pictures can be loaded with MemoryImage. This is an occasional situation, so a sample program cannot be provided. And next time you open the program, you can display it normally.

**Flutter doctor** 
`[✓] Flutter (Channel stable, 1.22.4, on Mac OS X 10.15.6 19G73 darwin-x64, locale zh-Hans-CN)
    • Flutter version 1.22.4 at /Users/57block/Development/src/flutter
    • Framework revision 1aafb3a8b9 (3 weeks ago), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/57block/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✗] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install:
        sudo gem install cocoapods

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0
`

**Comment by pedromassangocode** [comment]
Hi @Warpath 
Your code does not seems to reproduce the issue.

![gif](https://user-images.githubusercontent.com/67424416/101349937-63742680-388e-11eb-9f0d-9b27d1ea5d25.gif)

**Comment by Warpath** [comment]
Hi, @pedromassangocode . This is an occasional problem, so it takes a few more attempts to appear.

**Comment by csacchetti** [comment]
I also had the same problem. Then I realized that every time I restarted the iOS app it changed the name of a folder and therefore I realized that I had not to save the complete path but only the one that followed the documents folder. While the path of the documents folder I had to recreate it at the beginning after each reboot. This is how it all works.

local path example:
/Users/pincopallino/Library/Developer/CoreSimulator/Devices/280F0E7F-7A53-4110-ADB6-0DC2A4A04E4F/data/Containers/Data/Application/7EE7D9F1-65D9-49BE-A0F5-DA675FFEA842/Documents/myapp/audios/1615304726364.mp3

This is the folder that iOS changes every time: 7EE7D9F1-65D9-49BE-A0F5-DA675FFEA842

I hope it will be useful to someone
